search.go

```diff
@@ -64,69 +44,169 @@ func (c *SearchController) Show(ctx *app.ShowSearchContext) error {
 			}
 		}
 
+		pagingLinks := paginationLinks(
+			buildAbsoluteURL(ctx.RequestData),
+			"q="+ctx.Q,
+			offset,
+			limit,
+			count,
+			len(result))
+
 		response := app.SearchResponse{
-			Links: &app.PagingLinks{},
+			Links: &pagingLinks,
 			Meta:  &app.WorkItemListResponseMeta{TotalCount: count},
 			Data:  result,
 		}
+		return ctx.OK(&response)
+	})
 
-		// prev link
-		if offset > 0 && count > 0 {
-			var prevStart int
-			// we do have a prev link
-			if offset <= count {
-				prevStart = offset - limit
-			} else {
-				// the first range that intersects the end of the useful range
-				prevStart = offset - (((offset-count)/limit)+1)*limit
-			}
-			realLimit := limit
-			if prevStart < 0 {
-				// need to cut the range to start at 0
-				realLimit = limit + prevStart
-				prevStart = 0
-			}
-			prev := fmt.Sprintf("%s?q=%s&page[offset]=%d&page[limit]=%d", buildAbsoluteURL(ctx.RequestData), ctx.Q, prevStart, realLimit)
-			response.Links.Prev = &prev
-		}
+}
 
-		// next link
-		nextStart := offset + len(result)
-		if nextStart < count {
-			// we have a next link
-			next := fmt.Sprintf("%s?q=%s&page[offset]=%d&page[limit]=%d", buildAbsoluteURL(ctx.RequestData), ctx.Q, nextStart, limit)
-			response.Links.Next = &next
-		}
+// Users runs the user search action.
+func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
 
-		// first link
-		var firstEnd int
-		if offset > 0 {
-			firstEnd = offset % limit // this is where the second page starts
-		} else {
-			// offset == 0, first == current
-			firstEnd = limit
-		}
-		first := fmt.Sprintf("%s?q=%s&page[offset]=%d&page[limit]=%d", buildAbsoluteURL(ctx.RequestData), ctx.Q, 0, firstEnd)
-		response.Links.First = &first
-
-		// last link
-		var lastStart int
-		if offset < count {
-			// advance some pages until touching the end of the range
-			lastStart = offset + (((count - offset - 1) / limit) * limit)
+	q := ctx.Q
+	if q == "" {
```

Comment:
This seems arbitrary: we're doing a prefix search here. "" is the prefix of all strings. We don't allow "foo*" as a wildcard search either.